lib/iex/lib/iex/autocomplete.ex

```diff
@@ -297,26 +297,48 @@ defmodule IEx.Autocomplete do
       not ensure_loaded?(mod) ->
         []
       docs = Code.get_docs(mod, :docs) ->
-        exports(mod) |> Enum.reject(&hidden_fun?(&1, docs))
+        (exports(mod) -- get_default_arg_functions_with_doc_false(docs))
+        |> Enum.reject(&underscored_fun_without_docs?(&1, docs))
       true ->
         exports(mod)
     end
   end
 
-  defp hidden_fun?(fun, docs) do
+  defp get_default_arg_functions_with_doc_false(docs) do
+    Enum.reduce(docs, [], fn
+      {{fun_name, arity}, _, _, args, false}, acc ->
+        count =
+          Enum.count(args, fn
+            {:\\, _, _} -> true
+            _ -> false
+          end)
+
+        if count > 0 do
```

Comment:
We should remove this clause because even if it has no defaults, we want to remove at least the current arity.